packages/fx-core/src/plugins/resource/bot/errors.ts

```diff
@@ -78,10 +116,14 @@ export class SomethingMissingError extends PreconditionError {
   }
 }
 
-export function CheckThrowSomethingMissing(name: string, value: any): void {
+export function CheckThrowSomethingMissing<T>(
```

Comment:
Do we still need this method? Is there any possible to leverage below method because there are two methods with similar name which may confused people.

Reply:
this method use type guard like:
```
if (CheckThrowSomethingMissing("name", value)) {
  // then value will not be undifined
}
```

the second one can be used like 
```
// value will not be undifined
let value = CheckAndThrowIfMissing("name", value);
```

different action, the same effect